README.md

```diff
@@ -136,3 +136,22 @@ rules:
           name_regex: ".*"
           display_name_regex: "^!.+$"
 ```
+
+
+### prevent-tasks-with-tag-on-variants
+Prevent tasks with specific tags from being used in buildvariants with certain tags.
+
+#### Configuration
+
+Which tasks should be prevented in which variants can be configured via the lint config files. The
+configuration should use the following format:
+
+
+```yaml
+rules:
+  - rule: "prevent-tasks-with-tag-on-variants"
+    tags:
+      - variant_tag_name: "no_task_tag_experimental"
+        prevent_task_tag: "experimental"
```

Comment:
I will update the name to be more clear. 

> Also, does this need to generalize as an array? forbidden_task_tags (plural)?

We cannot use an array for prevent_task_tag because each variant_tag_name, prevent_task_tag, and ignored_tasks form a specific, interconnected set. The variant_tag_name (e.g., "no_task_tag_experimental") directly corresponds to a specific prevent_task_tag (e.g., "experimental") and also with a specific list of ignored_tasks. If we need to prevent a different tag, like "critical", it would require a separate tag with its own corresponding variant_tag_name (e.g., "no_task_tag_critical") and new list of ignored_tasks.